### ISSUE TITLE
Always redirecting to iOS App Store

### ISSUE THREAD

**Issue by bismarabia** [GeekyAnts/external_app_launcher#4]
I have defined random scheme in Project 1 (left side), called it '**dashboss**'
and tried to use this plugin to check for being installed and launch it.

![image](https://user-images.githubusercontent.com/16321223/147248123-e4c5e59f-5682-4220-94a5-ae0a40619637.png)

But I am unable to launch the app and it keeps saying **Redirecting to AppStore as the app is not present on the device**

What could be the issue?

**Comment by sarika01** [comment]
Have you added
`<key>LSApplicationQueriesSchemes</key>
<array>
  <string>dashboss</string> // url scheme name of the app
</array>`

in the info.plist file?

**Comment by bismarabia** [comment]
Yes I did @sarika01 

but still getting this the following logs

```
flutter: installed -> 0
flutter: App not found in the device
```

**Comment by bismarabia** [comment]
all of tests are on the simulator btw...

**Comment by bismarabia** [comment]
@sarika01 any update on this? I need this for an urgent task...

**Comment by GayathrideviGA** [comment]
@bismarabia 
Can you check for ios app store link once ? 

Also try to set
`openStore:false`
In main.dart file  and test the case.

**Comment by bismarabia** [comment]
@GayathrideviGA @sarika01 

it still says `flutter: App not found in the device`

this is from app I want to launch.
```xml
<key>LSApplicationQueriesSchemes</key>
<array>
    <string>pulsesecure</string>
</array>
```

and this is from the app I'm launching from
```dart
await LaunchApp.openApp(
    androidPackageName: '{packageID}',
    iosUrlScheme: 'pulsesecure://',
);
```


I tried this in iOS from a simulator (debug mode).

**Comment by bismarabia** [comment]
@sarika01 @GayathrideviGA  any update on this?

**Comment by Gayathri-GA** [comment]
@bismarabia 
Try to check the case with a real device as some times simulator could not able to open up App Store link it results app not found exception. Also it seems it works well with android device so try to debug it with real device with `openStore: true` .

**Comment by keylinwu-wmg** [comment]
@GayathrideviGA  This issue keeps happening, seems like the library is only working `Android`, but not working on `iOS`. 
@bismarabia Were you able to fix it in the end, I would really appreciate some help on this, otherwise, I will need to find another library :/

**Comment by bismarabia** [comment]
Hi @keylinwu-wmg

It works fine on Android. I tested it....
for iPhone. it doesn't work on simulator at all. I tested it on a real iPhone release mode and it always directs to AppStore even thought the app is installed.

For me, it's ok, at least at this point, I'm gonna leave it like this since the user can open the app from AppStore, it will say "OPEN"  if installed, "GET" otherwise.

Looking forward to seeing a fix for this issue. @jaiswalshubham84 @Gayathri-GA

**Comment by kha1989led** [comment]
To overcome this issue, I'm using this package (2.0.2) along with [url_launcher 6.0.18](https://pub.dev/packages/url_launcher):

```
if (Platform.isAndroid) {
  if (await LaunchApp.isAppInstalled(androidPackageName: appId)) {
    await LaunchApp.openApp(
        androidPackageName: appId
    );
  } else {
    launch(storeUrl);
  }
} else if (Platform.isIOS) {
  if (await canLaunch(appId)) {
    launch(appId);
  } else {
    launch(storeUrl);
  }
}
```

**Comment by Gayathri-GA** [comment]
The support for Android V2 embedding is now available in external_app_launcher: ^3.0.0
